### DISCUSSION TITLE
adding SNPs

### DISCUSSION THREAD

**Discussion by david-cros** [tpook92/MoBPS#5] [Q&A] [answered]
Hello
When importing a VCF file to generate an initial population (with create.diploid) is there a way to get more SNPs than what we have in the VCF ? I am not sure about what add.chromosome can do, but I think this is for something else.
Thanks
David C

**Comment by tpook92** [comment, marked as answer]
Dear David,

you question is actually very similar to a question ask during our last MoBPS Workshop. Please have a look at https://github.com/tpook92/MoBPS/blob/master/MoBPS_Workshop_GFT%26KWS/Workshop_additional_examples.R . lines 101ff. In this example a VCF file with 5000 SNPs is imported and subsequently  5000 additional SNPs are added to chromosome 1. 

Note that the additionally added SNPs were all simulated to be not linked to any of the SNPs from the VCF file - a method for this does not existed within MoBPS. If you were looking for a method to do that I would suggest to sample your additional variants from a correlated random variable to at least get tome LD. Something potentially more realistic that should also be able to grasp linkage would be the use of long range haplotype blocks from a software like HaploBlocker (https://github.com/tpook92/HaploBlocker). Or what did you have in mind here?

Add.chromosome is a feature to add one chromosome. This is mostly used internally, but not required unless you have separate VCF files from each chromosome that you do not want to merge before importing to MoBPS.

Best regards 
Torsten

**Reply by david-cros** [reply]
Thanks for the explanations. My point was indeed to add extras SNPs that would respect the LD profile. I will try to improve my VCF as you suggested then.
David
